### ISSUE TITLE
Timeout error on gaf-eco-mapping-derived.txt

### ISSUE THREAD

**Issue by dmiers** [evidenceontology/evidenceontology#250]
MGI downloads the gaf-eco-mapping-derived.txt file each evening via wget. Starting last night, our download process just gets a timeout error while trying to grab the file:

--2020-03-20 06:21:23--  https://raw.githubusercontent.com/evidenceontology/evidenceontology/master/gaf-eco-mapping-derived.txt
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 199.232.36.133
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|199.232.36.133|:443... failed: Connection timed out.

Any idea what the problem is?

**Comment by beckyjackson** [comment]
Perhaps GitHub is having issues when you ran it? I just tested `wget` on that link and was able to successfully download it:

```
(base) Rebeccas-MacBook-Pro-2:IHCC tauber$ wget https://raw.githubusercontent.com/evidenceontology/evidenceontology/master/gaf-eco-mapping-derived.txt
--2020-03-20 06:44:11--  https://raw.githubusercontent.com/evidenceontology/evidenceontology/master/gaf-eco-mapping-derived.txt
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.52.133
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.52.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 19449 (19K) [text/plain]
Saving to: ‘gaf-eco-mapping-derived.txt’

gaf-eco-mapping-der 100%[===================>]  18.99K  --.-KB/s    in 0.02s   

2020-03-20 06:44:11 (1019 KB/s) - ‘gaf-eco-mapping-derived.txt’ saved [19449/19449]
```

Could you try running it again? If you're still having problems, could it be something with your server?

**Comment by dmiers** [comment]
I'll check with our IT department and see if anything has changed on our end because wget still won't work.

**Comment by dmiers** [comment]
It turns out that it's a network security issue on our end and it is in the process of being resolved.

**Comment by cmungall** [comment]
I recommend avoiding hardcoding the URL, always use the PURL: http://purl.obolibrary.org/obo/eco/gaf-eco-mapping-derived.txt

@beckyjackson can we have the header of the file include metadata including the preferred PURL, as we have here:

```
$ curl -L -s http://purl.obolibrary.org/obo/eco/gaf-eco-mapping.txt | head -20
# Permanent URL of this document:
#  http://purl.obolibrary.org/obo/eco/gaf-eco-mapping.txt
#
# Always use this URL
#
# About:
# This mapping translates evidence codes used in a GAF to a specific evidence type in ECO
#
# Each mapping is from a pair:
#  1. GAF evidence code (e.g. IEA)
#  2. A GO_REF (found in the GAF Reference column; see http://www.geneontology.org/cgi-bin/references.cgi)
# to a class identifier in ECO (http://purl.obolibrary.org/obo/eco.owl)
#  3. ECO ID
#
# If the value 'Default' is in column 2 then this mapping is used if
# no more specific mapping can be found
#
# This document consists of these comments (after #s) followed by
# a set of mappings, each represented as the above 3 columns separated by tabs
#
```

**Comment by beckyjackson** [comment]
I think this issue is resolved, and the header is taken care of in #249
